packages/shared/src/guards.ts

```diff
@@ -412,11 +423,13 @@ export function isComplexValue(
   return type.class === DTypeClass.Complex;
 }
 
-function assertScalarValue(
```

Comment:
Swapping the two parameters to match `assertArrayValue` and have the `value` being asserted first.